### ISSUE TITLE
No module named 'urlparse' and No module named 'StringIO'

### ISSUE THREAD

**Issue by TheRoarkster** [tjfontaine/airprint-generate#20]
This is an old, un-updated script that lots of people still use but has not kept up with python changes.  If you get either of the above errors you need to just:

```
cd /opt/airprint
sudo nano airprint-generate.py
```
[Edit]
```
#import cups, os, optparse, re, urlparse
import cups, os, optparse, re, urllib.parse
```
[Edit]

```
#from StringIO import StringIO
from io import StringIO
```
I would just try to branch, but I am not a skilled programmer and just found these edits through searches.

**Comment by branx86** [comment]
I have CentOs 8 and did the Edit import cups, os, optparse, re, urllib.parse  ,  from io import StringIO
Now I have : File "./airprint-generate.py", line 277, in <module>
    apg.generate()
  File "./airprint-generate.py", line 127, in generate
    uri = urlparse.urlparse(v['printer-uri-supported'])
NameError: name 'urlparse' is not defined

**Comment by branx86** [comment]
Is their missing depend or do I have the wrong Python version Using (/usr/bin/Python3)

**Comment by branx86** [comment]
Well figured out the above problem by Doing in CentOs 8  " pip2 install cups" and " pip3 install cups" now I have :  File "./airprint-generate.py", line 261, in <module>
    cups.setPasswordCB(getpass)
AttributeError: module 'cups' has no attribute 'setPasswordCB'

**Comment by tjfontaine** [comment]
much of this has been fixed in 0384693e462ee6a0bf9237a839d73d4c6a720cd3 though I would suggest using the docker container that you can build from the repo.
